src/riot-rs-embassy/src/lib.rs

```diff
@@ -208,10 +228,10 @@ async fn init_task(peripherals: arch::Peripherals) {
         let usb_config = usb_default_config();
 
         #[cfg(context = "nrf52")]
-        let usb_driver = nrf52::usb::driver(peripherals.USBD);
+        let usb_driver = nrf52::usb::driver(peripherals.USBD.take().unwrap());
```

Comment:
Being able to do do `peripherals.X.take()` should make it possible to change `driver(x: USBD)` into `driver(p: &mut OptionalPeripherals)`, and turn these two cases into one.